examples/discover/main.go

```diff
@@ -0,0 +1,319 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/containerd/containerd/remotes"
+	imagespec "github.com/opencontainers/image-spec/specs-go/v1"
+	artifactspec "github.com/oras-project/artifacts-spec/specs-go/v1"
+	"oras.land/oras-go/pkg/artifact"
+	"oras.land/oras-go/pkg/content"
+	"oras.land/oras-go/pkg/oras"
+	"oras.land/oras-go/pkg/remotes/docker"
+)
+
+const (
+	repoRef  = "localhost:5000/net-monitor"
+	imageRef = "localhost:5000/net-monitor:v1"
+)
+
+var (
+	localhostResolver  remotes.Resolver
+	example_references []artifactspec.Descriptor
+	targetResolver     *content.Memory
+)
+
+func main() {
+	ctx := context.Background()
+
+	registry, err := content.NewRegistry(content.RegistryOptions{PlainHTTP: true})
+	if err != nil {
+		panic("could not create a new registry")
+	}
+	discoverer, err := docker.WithDiscover(imageRef, localhostResolver, &registry.Opts)
+	if err != nil {
+		panic("could not create discoverer")
+	}
+
+	desc, blobs, err := oras.Discover(ctx, discoverer, imageRef, "")
```

Comment:
`Discover` discovers referrer manifests referencing `desc`. They are `manifests`, not `blobs`.

Reply:
Clarify what you want me to change please